File: releases/v2.1.0/schemaorgphp/classes/job_posting.php
<?php
namespace SchemaOrg;

// JobPosting see : https://schema.org/JobPosting
class JobPosting implements \JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type    = 'JobPosting';

	/**
	 * With properties from Intangible see : https://schema.org/Intangible
	 */

	/**
	 * With properties from Thing see : https://schema.org/Thing
	 */


	/**
	 * An additional type for the item, typically used for adding more specific types from external vocabularies in microdata syntax. This is a relationship between something and a class that the thing is in. In RDFa syntax, it is better to use the native RDFa syntax - the &#39;typeof&#39; attribute - for multiple types. Schema.org tools may have only weaker understanding of extra types, in particular those defined externally.
	 * see : https://schema.org/additionalType
	 *
	 * @var string | string[]
	 */
	public $additional_type;

	/**
	 * An alias for the item.
	 * see : https://schema.org/alternateName
	 *
	 * @var string | string[]
	 */
	public $alternate_name;

	/**
	 * The base salary of the job or of an employee in an EmployeeRole.
	 * see : https://schema.org/baseSalary
	 *
	 * @var float | float[] | \PriceSpecification | \PriceSpecification[]
	 */
	public $base_salary;

	/**
	 * Description of benefits associated with the job.
	 * see : https://schema.org/benefits
	 *
	 * @var string | string[]
	 */
	public $benefits;

	/**
	 * Publication date for the job posting.
	 * see : https://schema.org/datePosted
	 *
	 * @var string | string[]
	 */
	public $date_posted;

	/**
	 * A short description of the item.
	 * see : https://schema.org/description
	 *
	 * @var string | string[]
	 */
	public $description;

	/**
	 * Educational background needed for the position.
	 * see : https://schema.org/educationRequirements
	 *
	 * @var string | string[]
	 */
	public $education_requirements;

	/**
	 * Type of employment (e.g. full-time, part-time, contract, temporary, seasonal, internship).
	 * see : https://schema.org/employmentType
	 *
	 * @var string | string[]
	 */
	public $employment_type;

	/**
	 * Description of skills and experience needed for the position.
	 * see : https://schema.org/experienceRequirements
	 *
	 * @var string | string[]
	 */
	public $experience_requirements;

	/**
	 * Organization offering the job position.
	 * see : https://schema.org/hiringOrganization
	 *
	 * @var \Organization | \Organization[]
	 */
	public $hiring_organization;

	/**
	 * An image of the item. This can be a &lt;a href=&quot;http://schema.org/URL&quot;&gt;URL&lt;/a&gt; or a fully described &lt;a href=&quot;http://schema.org/ImageObject&quot;&gt;ImageObject&lt;/a&gt;.
	 * see : https://schema.org/image
	 *
	 * @var string | string[] | \ImageObject | \ImageObject[]
	 */
	public $image;

	/**
	 * Description of bonus and commission compensation aspects of the job.
	 * see : https://schema.org/incentiveCompensation
	 *
	 * @var string | string[]
	 */
	public $incentive_compensation;

	/**
	 * Description of bonus and commission compensation aspects of the job.
	 * see : https://schema.org/incentives
	 *
	 * @var string | string[]
	 */
	public $incentives;

	/**
	 * The industry associated with the job position.
	 * see : https://schema.org/industry
	 *
	 * @var string | string[]
	 */
	public $industry;

	/**
	 * Description of benefits associated with the job.
	 * see : https://schema.org/jobBenefits
	 *
	 * @var string | string[]
	 */
	public $job_benefits;

	/**
	 * A (typically single) geographic location associated with the job position.
	 * see : https://schema.org/jobLocation
	 *
	 * @var \Place | \Place[]
	 */
	public $job_location;

	/**
	 * Indicates a page (or other CreativeWork) for which this thing is the main entity being described.
	 *       &lt;br /&gt;&lt;br /&gt;
	 *       See &lt;a href=&quot;/docs/datamodel.html#mainEntityBackground&quot;&gt;background notes&lt;/a&gt; for details.
	 *
	 * see : https://schema.org/mainEntityOfPage
	 *
	 * @var \CreativeWork | \CreativeWork[] | string | string[]
	 */
	public $main_entity_of_page;

	/**
	 * The name of the item.
	 * see : https://schema.org/name
	 *
	 * @var string | string[]
	 */
	public $name;

	/**
	 * Category or categories describing the job. Use BLS O*NET-SOC taxonomy: http://www.onetcenter.org/taxonomy.html. Ideally includes textual label and formal code, with the property repeated for each applicable value.
	 * see : https://schema.org/occupationalCategory
	 *
	 * @var string | string[]
	 */
	public $occupational_category;

	/**
	 * Indicates a potential Action, which describes an idealized action in which this thing would play an &#39;object&#39; role.
	 * see : https://schema.org/potentialAction
	 *
	 * @var \Action | \Action[]
	 */
	public $potential_action;

	/**
	 * Specific qualifications required for this role.
	 * see : https://schema.org/qualifications
	 *
	 * @var string | string[]
	 */
	public $qualifications;

	/**
	 * Responsibilities associated with this role.
	 * see : https://schema.org/responsibilities
	 *
	 * @var string | string[]
	 */
	public $responsibilities;

	/**
	 * The currency (coded using ISO 4217, http://en.wikipedia.org/wiki/ISO_4217 ) used for the main salary information in this job posting or for this employee.
	 * see : https://schema.org/salaryCurrency
	 *
	 * @var string | string[]
	 */
	public $salary_currency;

	/**
	 * URL of a reference Web page that unambiguously indicates the item&#39;s identity. E.g. the URL of the item&#39;s Wikipedia page, Freebase page, or official website.
	 * see : https://schema.org/sameAs
	 *
	 * @var string | string[]
	 */
	public $same_as;

	/**
	 * Skills required to fulfill this role.
	 * see : https://schema.org/skills
	 *
	 * @var string | string[]
	 */
	public $skills;

	/**
	 * Any special commitments associated with this job posting. Valid entries include VeteranCommit, MilitarySpouseCommit, etc.
	 * see : https://schema.org/specialCommitments
	 *
	 * @var string | string[]
	 */
	public $special_commitments;

	/**
	 * The title of the job.
	 * see : https://schema.org/title
	 *
	 * @var string | string[]
	 */
	public $title;

	/**
	 * URL of the item.
	 * see : https://schema.org/url
	 *
	 * @var string | string[]
	 */
	public $url;

	/**
	 * The typical working hours for this job (e.g. 1st shift, night shift, 8am-5pm).
	 * see : https://schema.org/workHours
	 *
	 * @var string | string[]
	 */
	public $work_hours;

	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type'    => 'JobPosting',
		);

		$serialized = \SchemaOrg\json_serialize( $this->additional_type );
		if ( ! empty( $serialized ) ) {
			$out['additionalType'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->alternate_name );
		if ( ! empty( $serialized ) ) {
			$out['alternateName'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->base_salary );
		if ( ! empty( $serialized ) ) {
			$out['baseSalary'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->benefits );
		if ( ! empty( $serialized ) ) {
			$out['benefits'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->date_posted );
		if ( ! empty( $serialized ) ) {
			$out['datePosted'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->description );
		if ( ! empty( $serialized ) ) {
			$out['description'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->education_requirements );
		if ( ! empty( $serialized ) ) {
			$out['educationRequirements'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->employment_type );
		if ( ! empty( $serialized ) ) {
			$out['employmentType'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->experience_requirements );
		if ( ! empty( $serialized ) ) {
			$out['experienceRequirements'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->hiring_organization );
		if ( ! empty( $serialized ) ) {
			$out['hiringOrganization'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->image );
		if ( ! empty( $serialized ) ) {
			$out['image'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->incentive_compensation );
		if ( ! empty( $serialized ) ) {
			$out['incentiveCompensation'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->incentives );
		if ( ! empty( $serialized ) ) {
			$out['incentives'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->industry );
		if ( ! empty( $serialized ) ) {
			$out['industry'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->job_benefits );
		if ( ! empty( $serialized ) ) {
			$out['jobBenefits'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->job_location );
		if ( ! empty( $serialized ) ) {
			$out['jobLocation'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->main_entity_of_page );
		if ( ! empty( $serialized ) ) {
			$out['mainEntityOfPage'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->name );
		if ( ! empty( $serialized ) ) {
			$out['name'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->occupational_category );
		if ( ! empty( $serialized ) ) {
			$out['occupationalCategory'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->potential_action );
		if ( ! empty( $serialized ) ) {
			$out['potentialAction'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->qualifications );
		if ( ! empty( $serialized ) ) {
			$out['qualifications'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->responsibilities );
		if ( ! empty( $serialized ) ) {
			$out['responsibilities'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->salary_currency );
		if ( ! empty( $serialized ) ) {
			$out['salaryCurrency'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->same_as );
		if ( ! empty( $serialized ) ) {
			$out['sameAs'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->skills );
		if ( ! empty( $serialized ) ) {
			$out['skills'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->special_commitments );
		if ( ! empty( $serialized ) ) {
			$out['specialCommitments'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->title );
		if ( ! empty( $serialized ) ) {
			$out['title'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->url );
		if ( ! empty( $serialized ) ) {
			$out['url'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->work_hours );
		if ( ! empty( $serialized ) ) {
			$out['workHours'] = $serialized;
		}

		return $out;
	}
}

File: releases/v3.2.0/schemaorggo/job_posting.go
package schemaorggo

import "encoding/json"

// JobPosting see : https://schema.org/JobPosting
type JobPosting struct {

	// With properties from Intangible see : https://schema.org/Intangible
	//

	// With properties from Thing see : https://schema.org/Thing
	//

	// AdditionalType see : https://schema.org/additionalType
	// An additional type for the item, typically used for adding more specific types from external vocabularies in microdata syntax. This is a relationship between something and a class that the thing is in. In RDFa syntax, it is better to use the native RDFa syntax - the &#39;typeof&#39; attribute - for multiple types. Schema.org tools may have only weaker understanding of extra types, in particular those defined externally.
	// types : URL
	AdditionalType []string `json:"additionalType,omitempty"`

	// AlternateName see : https://schema.org/alternateName
	// An alias for the item.
	// types : Text
	AlternateName []string `json:"alternateName,omitempty"`

	// BaseSalary see : https://schema.org/baseSalary
	// The base salary of the job or of an employee in an EmployeeRole.
	// types : Number PriceSpecification MonetaryAmount
	BaseSalary []interface{} `json:"baseSalary,omitempty"`

	// Benefits see : https://schema.org/benefits
	// Description of benefits associated with the job.
	// types : Text
	Benefits []string `json:"benefits,omitempty"`

	// DatePosted see : https://schema.org/datePosted
	// Publication date for the job posting.
	// types : Date
	DatePosted []Date `json:"datePosted,omitempty"`

	// Description see : https://schema.org/description
	// A description of the item.
	// types : Text
	Description []string `json:"description,omitempty"`

	// DisambiguatingDescription see : https://schema.org/disambiguatingDescription
	// A sub property of description. A short description of the item used to disambiguate from other, similar items. Information from other properties (in particular, name) may be necessary for the description to be useful for disambiguation.
	// types : Text
	DisambiguatingDescription []string `json:"disambiguatingDescription,omitempty"`

	// EducationRequirements see : https://schema.org/educationRequirements
	// Educational background needed for the position.
	// types : Text
	EducationRequirements []string `json:"educationRequirements,omitempty"`

	// EmploymentType see : https://schema.org/employmentType
	// Type of employment (e.g. full-time, part-time, contract, temporary, seasonal, internship).
	// types : Text
	EmploymentType []string `json:"employmentType,omitempty"`

	// ExperienceRequirements see : https://schema.org/experienceRequirements
	// Description of skills and experience needed for the position.
	// types : Text
	ExperienceRequirements []string `json:"experienceRequirements,omitempty"`

	// HiringOrganization see : https://schema.org/hiringOrganization
	// Organization offering the job position.
	// types : Organization
	HiringOrganization []*Organization `json:"hiringOrganization,omitempty"`

	// Identifier see : https://schema.org/identifier
	// The identifier property represents any kind of identifier for any kind of [[Thing]], such as ISBNs, GTIN codes, UUIDs etc. Schema.org provides dedicated properties for representing many of these, either as textual strings or as URL (URI) links. See [background notes](/docs/datamodel.html#identifierBg) for more details.
	//
	// types : URL Text PropertyValue
	Identifier []interface{} `json:"identifier,omitempty"`

	// Image see : https://schema.org/image
	// An image of the item. This can be a [[URL]] or a fully described [[ImageObject]].
	// types : URL ImageObject
	Image []interface{} `json:"image,omitempty"`

	// IncentiveCompensation see : https://schema.org/incentiveCompensation
	// Description of bonus and commission compensation aspects of the job.
	// types : Text
	IncentiveCompensation []string `json:"incentiveCompensation,omitempty"`

	// Incentives see : https://schema.org/incentives
	// Description of bonus and commission compensation aspects of the job.
	// types : Text
	Incentives []string `json:"incentives,omitempty"`

	// Industry see : https://schema.org/industry
	// The industry associated with the job position.
	// types : Text
	Industry []string `json:"industry,omitempty"`

	// JobBenefits see : https://schema.org/jobBenefits
	// Description of benefits associated with the job.
	// types : Text
	JobBenefits []string `json:"jobBenefits,omitempty"`

	// JobLocation see : https://schema.org/jobLocation
	// A (typically single) geographic location associated with the job position.
	// types : Place
	JobLocation []*Place `json:"jobLocation,omitempty"`

	// MainEntityOfPage see : https://schema.org/mainEntityOfPage
	// Indicates a page (or other CreativeWork) for which this thing is the main entity being described. See [background notes](/docs/datamodel.html#mainEntityBackground) for details.
	// types : CreativeWork URL
	MainEntityOfPage []interface{} `json:"mainEntityOfPage,omitempty"`

	// Name see : https://schema.org/name
	// The name of the item.
	// types : Text
	Name []string `json:"name,omitempty"`

	// OccupationalCategory see : https://schema.org/occupationalCategory
	// Category or categories describing the job. Use BLS O*NET-SOC taxonomy: http://www.onetcenter.org/taxonomy.html. Ideally includes textual label and formal code, with the property repeated for each applicable value.
	// types : Text
	OccupationalCategory []string `json:"occupationalCategory,omitempty"`

	// PotentialAction see : https://schema.org/potentialAction
	// Indicates a potential Action, which describes an idealized action in which this thing would play an &#39;object&#39; role.
	// types : Action
	PotentialAction []*Action `json:"potentialAction,omitempty"`

	// Qualifications see : https://schema.org/qualifications
	// Specific qualifications required for this role.
	// types : Text
	Qualifications []string `json:"qualifications,omitempty"`

	// Responsibilities see : https://schema.org/responsibilities
	// Responsibilities associated with this role.
	// types : Text
	Responsibilities []string `json:"responsibilities,omitempty"`

	// SalaryCurrency see : https://schema.org/salaryCurrency
	// The currency (coded using [ISO 4217](http://en.wikipedia.org/wiki/ISO_4217) ) used for the main salary information in this job posting or for this employee.
	// types : Text
	SalaryCurrency []string `json:"salaryCurrency,omitempty"`

	// SameAs see : https://schema.org/sameAs
	// URL of a reference Web page that unambiguously indicates the item&#39;s identity. E.g. the URL of the item&#39;s Wikipedia page, Wikidata entry, or official website.
	// types : URL
	SameAs []string `json:"sameAs,omitempty"`

	// Skills see : https://schema.org/skills
	// Skills required to fulfill this role.
	// types : Text
	Skills []string `json:"skills,omitempty"`

	// SpecialCommitments see : https://schema.org/specialCommitments
	// Any special commitments associated with this job posting. Valid entries include VeteranCommit, MilitarySpouseCommit, etc.
	// types : Text
	SpecialCommitments []string `json:"specialCommitments,omitempty"`

	// Title see : https://schema.org/title
	// The title of the job.
	// types : Text
	Title []string `json:"title,omitempty"`

	// Url see : https://schema.org/url
	// URL of the item.
	// types : URL
	Url []string `json:"url,omitempty"`

	// ValidThrough see : https://schema.org/validThrough
	// The date after when the item is not valid. For example the end of an offer, salary period, or a period of opening hours.
	// types : DateTime
	ValidThrough []DateTime `json:"validThrough,omitempty"`

	// WorkHours see : https://schema.org/workHours
	// The typical working hours for this job (e.g. 1st shift, night shift, 8am-5pm).
	// types : Text
	WorkHours []string `json:"workHours,omitempty"`
}

func (v JobPosting) MarshalJSON() ([]byte, error) {
	type Alias JobPosting

	b, err := json.Marshal((Alias)(v))
	if err != nil {
		return nil, err
	}

	return append([]byte("{\"@context\":\"http://schema.org\",\"@type\":\"JobPosting\","), b[1:]...), nil
}
